Composer/packages/client/src/pages/publish/createPublishTarget.tsx

```diff
@@ -83,6 +84,7 @@ const CreatePublishTarget: React.FC<CreatePublishTargetProps> = (props) => {
     if (targetType) {
       await props.updateSettings(name, targetType, JSON.stringify(config) || '{}');
       props.closeDialog();
+      await filePersistence.flush();
```

Comment:
I'm not sure I understand the bug that well, but is it possible to do this somewhere else instead of the component? The file persistence layer needs to remain an implementation detail to components. This violates that principle.

Reply:
move it into getHistory action.